### ISSUE TITLE
Throwing an error from inside of `onExecute` should be handled as an execution error

### ISSUE THREAD

**Issue by ben-pr-p** [graphql-hive/envelop#918]
**Describe the bug**

If inside of an `onExecute` plugin, I do:
```typescript
export const myExecutePlugin = (): Plugin<MyContextType> => {
  return {
    onExecute: async ({ executeFn, setExecuteFn }) => {
      const newExecuteFn = async (args: TypedExecutionArgs<MyContextType>) => {
        throw new Error('Oh no! I couldnt execute');
      };

      setExecuteFn(newExecuteFn);
    },
  };
};
```

That error will be thrown all the way to the caller that called `getEnveloped().execute`.

**To Reproduce**

Code above

**Expected behavior**

Most of the time, the code that developers write will not expect `.execute` to potentially error, and this expectation is built into packages like `graphql-ws` and other GraphQL servers.

Instead, developers throughout the ecosystem have expected `result.errors` to be populated.

I propose that we wrap the final call to the executeFn wrapping all of the nested executeFn's in a try / catch statement here: https://github.com/dotansimha/envelop/blob/main/packages/core/src/orchestrator.ts#L463

And set `.errors` as expected.

**Comment by ben-pr-p** [comment]
In the meantime, here is a simple plugin for anyone else running into this issue to guarantee all errors will be handled as other libraries expect:
```typescript
import { Plugin } from '@envelop/types';
import { ExecutionResult } from 'graphql';

export const useHandleUnexpectedError: Plugin = {
  onExecute: async ({ executeFn, setExecuteFn }) => {
    const handleErrorsExecuteFn = async (args): Promise<ExecutionResult> => {
      try {
        const result = await executeFn(args);
        return result;
      } catch (ex) {
        return {
          errors: [ex],
        };
      }
    };

    setExecuteFn(handleErrorsExecuteFn);
  },
};
```

**Comment by n1ru4l** [comment]
This is interesting. The default `execute` function should actually never throw as far as I know. By doing this we would probably change the default GraphQL behavior. Why would you want to throw within execute instead of returning `{ errors: [new GraphQLError("...")]}`?

**Comment by ben-pr-p** [comment]
> The default execute function should actually never throw as far as I know

Yes, I think that's the problem. No other library is wrapping execute with try / catch, so we should.

> Why would you want to throw within execute instead of returning `{ errors: [new GraphQLError("...")]}`?

It wasn't on purpose :) 

I'm opening up this issue because I had an error that ended up causing `graphql-ws` to close the whole web socket with that error, and it took a very long time to track down where it was coming from because no one expected an error to be able to be produced there.

**Comment by ben-pr-p** [comment]
From the perspective of a first-time plugin author, I expected throwing to lead to a properly set `{ errors: [] }` object because that's the behavior I'm used to if I throw inside of a resolver. 

Of course, this is a different context and that expectation isn't totally founded. I'm happy now because I have the above plugin ^^^, but other users may also come into writing plugins with that same expectation

**Comment by n1ru4l** [comment]
That is interesting, in helix we wanna log such kind of errors to the console and map it to an unexpected error. graphql-ws could do something similar 🤔 Our libraries should not swallow errors but instead give useful information to developers. Not sure whether this should be the responsibility of envelop

**Comment by ben-pr-p** [comment]
`graphql-ws` handles it by sending the error to the client and closing the socket, but often times the error can't be encoded in a socket closing message so you get this: https://github.com/enisdenjo/graphql-ws/issues/73

I think swallowing errors in the status quo, and making the behavior I suggested the default prevents errors from getting swallowed.

I think the other thing to consider is that many `onExecute` plugins are actually doing things during execute (mapping over the result, etc.) and so an error there should be handled like any other execution error.
 
But trust your judgement here via Helix, etc. and appreciate envelop so much! Hopefully the snippet above I pasted can help someone else out as well
